Set filetype html for Vue.js files.

rillian opened the pull request:
Identify Vue.js Single File Component source files by their .vue
filename extension and set the filetype to html. They're a mixture
of html templates, css, and javascript. The html syntax highlighting
works well enough.

rillian committed:
Set filetype html for Vue.js files.

Identify Vue.js Single File Component source files by their .vue
filename extension and set the filetype to html. They're a mixture
of html templates, css, and javascript. The html syntax highlighting
works well enough.

brammool commented:
> Identify Vue.js Single File Component source files by their .vue
> filename extension and set the filetype to html. They're a mixture
> of html templates, css, and javascript. The html syntax highlighting
> works well enough.

That's not the right way.  What should happen is to detect *.vue files
as "vue" filetype, then add plugin files for vue that load the related
html plugin.  That way a user can override the plugin with something
more specific, if he wants to.

-- 
GUARD #2:  It could be carried by an African swallow!
GUARD #1:  Oh, yeah, an African swallow maybe, but not a European swallow,
           that's my point.
GUARD #2:  Oh, yeah, I agree with that...
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

rillian committed:
Use html syntax for Vue.js.

Instead of treating .vue files as html directly, define a `vuejs`
filetype and syntax file which wraps the html syntax file. This
makes it possible to override the default highlighting.

rillian commented:
Thanks for the quick review, @brammool. I've attempted to address your comments. Please let me know how I did. :)

brammool commented:
Ralph Giles wrote:

> Thanks for the quick review, @brammool. I've attempted to address your
> comments. Please let me know how I did. :)

Looks good, thanks.  I'll include it.

-- 
ARTHUR:  You fight with the strength of many men, Sir knight.
         I am Arthur, King of the Britons.  [pause]
         I seek the finest and the bravest knights in the land to join me
         in my Court of Camelot.  [pause]
         You have proved yourself worthy; will you join me?  [pause]
         You make me sad.  So be it.  Come, Patsy.
BLACK KNIGHT:  None shall pass.
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

rillian commented:
Lovely, thanks!

UncleBill commented:
Can we rename `vuejs` to `vue` instead? @chrisbra

rillian commented:
I don't have an opinion. I thought `vuejs` was less likely to conflict with other projects.

brammool commented:
> I don't have an opinion. I thought `vuejs` was less likely to conflict
> with other projects.

Also: the website is vuejs.org

-- 
Two cows are standing together in a field.  One asks the other:
"So what do you think about this Mad Cow Disease?"
The other replies: "That doesn't concern me. I'm a helicopter."

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

UncleBill commented:
>Also: the website is vuejs.org

Maybe because of `vue.org` is token. 

1. `.vue` is a single-file-component (SFC) which contains component's template, options, life-cycle hooks, and style, it more than a `js` file.

2. The most popular vim plugin for vue on Github, https://github.com/posva/vim-vue , uses `vue` as well.

adriaanzon commented:
I would be in favour of changing it to `vue`. There are already some plugins in the ecosystem that provide configuration for the `vue` filetype, e.g.:

* https://github.com/w0rp/ale/blob/master/ale_linters/vue/vls.vim
* https://github.com/Shougo/context_filetype.vim/blob/c3f806b4d06230607bc0a3bf619bf9c3ec1c9d95/autoload/context_filetype.vim#L170

It would be less of a hassle if existing plugins didn't have to change to `vuejs`.

Over at vim-vue this issue arose: https://github.com/posva/vim-vue/issues/138. If we decide to change vim-vue to use `vuejs` as filetype, it would break existing plugins that look for the `vue` filetype. So at this point I'm not really sure how to move forward with this.

brammool commented:
OK, let's call it "vue" then.
